/**:
  ros__parameters:

    # Topics
    pointCloudTopic: "/carla/lidar"              # Carla publishes raw LiDAR here
    imuTopic: "/carla/imu_sensor"                       # Carla IMU topic
    odomTopic: "/odometry/imu"                   # Will be published by LIO-SAM
    gpsTopic: "/odometry/gps"                    # If GNSS is available, navsat_transform_node can publish this

    # Frames
    lidarFrame: "lidar_link"
    baselinkFrame: "base_link"
    odometryFrame: "odom"
    mapFrame: "map"

    # GPS Settings (keep default unless using GNSS data)
    useImuHeadingInitialization: false
    useGpsElevation: false
    gpsCovThreshold: 2.0
    poseCovThreshold: 25.0

    # Export settings
    savePCD: false
    savePCDDirectory: "/Downloads/LOAM/"

    # Sensor Settings (adjusted for Carla)
    sensor: ouster                                # Carla simulated Ouster-like LIDAR
    N_SCAN: 64                                    # Carla default is usually 64
    Horizon_SCAN: 1024                            # Carla LiDAR res: adjust to match your Carla config
    downsampleRate: 1
    lidarMinRange: 1.0
    lidarMaxRange: 100.0

    # IMU Settings (keep default)
    imuAccNoise: 3.9939570888238808e-03
    imuGyrNoise: 1.5636343949698187e-03
    imuAccBiasN: 6.4356659353532566e-05
    imuGyrBiasN: 3.5640318696367613e-05
    imuGravity: 9.80511
    imuRPYWeight: 0.01

    # Extrinsics (adjust if lidar <-> imu transform in Carla is known)
    extrinsicTrans:  [ 0.0,  0.0,  0.0 ]
    extrinsicRot:    [-1.0,  0.0,  0.0,
                       0.0,  1.0,  0.0,
                       0.0,  0.0, -1.0 ]
    extrinsicRPY: [ 0.0,  1.0,  0.0,
                   -1.0,  0.0,  0.0,
                    0.0,  0.0,  1.0 ]

    # LOAM feature threshold
    edgeThreshold: 1.0
    surfThreshold: 0.1
    edgeFeatureMinValidNum: 10
    surfFeatureMinValidNum: 100

    # voxel filter paprams
    odometrySurfLeafSize: 0.4
    mappingCornerLeafSize: 0.2
    mappingSurfLeafSize: 0.4

    # robot motion constraint
    z_tollerance: 1000.0
    rotation_tollerance: 1000.0

    # CPU Params
    numberOfCores: 4
    mappingProcessInterval: 0.15

    # Surrounding map
    surroundingkeyframeAddingDistThreshold: 1.0
    surroundingkeyframeAddingAngleThreshold: 0.2
    surroundingKeyframeDensity: 2.0
    surroundingKeyframeSearchRadius: 50.0

    # Loop closure
    loopClosureEnableFlag: true
    loopClosureFrequency: 1.0
    surroundingKeyframeSize: 50
    historyKeyframeSearchRadius: 15.0
    historyKeyframeSearchTimeDiff: 30.0
    historyKeyframeSearchNum: 25
    historyKeyframeFitnessScore: 0.3

    # Visualization
    globalMapVisualizationSearchRadius: 1000.0
    globalMapVisualizationPoseDensity: 10.0
    globalMapVisualizationLeafSize: 1.0
